Sources/StreamChatUI/Utils/MarkdownFormatter.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftyMarkdown
+import UIKit
+
+public protocol MarkdownFormatter {
+    /// Checks for Markdown patterns in the given String.
+    /// - Parameter text: The string in which Markdown patters are going to be sought.
+    /// - Returns: Returns a Boolean value that indicates whether Markdown patters where found in the given String.
+    func containsMarkdown(text: String) -> Bool
+    /// Returns an attributed string form the given Markdown-formatted string.
+    /// - Parameter string: The string to be formatted.
+    /// - Returns: An attributed string with the corresponding formatted attributes.
+    func format(from string: String) -> NSAttributedString
+}
+
+/// Default implementation for the Markdown formatter
+class DefaultMarkdownFormatter: MarkdownFormatter {
```

Comment:
How about making the default implementation `open` so integrators can subclass and reuse the existed logic if they want?

Reply:
I wanted to check about this with you guys. Inheritance makes more sense for UI components that inherit from UIView - UIViewController but this is more of a formatting class. I used the `MarkdownFormatter` protocol because  commonly for Markdown support, similar as we are doing, integrators would probably use a 3rd party library which I think is best if they create their own abstraction wrapper class around that framework, using the `MarkdownFormatter` protocol so they can integrate it in the `Components.swift` class. But let me know if you still consider the inheritance approach better for consistency; in that case I'd better remove the protocol and probably rename the class removing the `Default` word and marking it as `open` as you suggested.

Reply:
The use-case I have in mind is a customer who knows that all messages contain a markdown, let's say. In that case, they would override `conatinsMarkdown` to always return `true` (to not run the regex and save time) but keep the default formatting. That's quite an edge case but there can be other scenarios when an integrator just wants to slightly tweak the default logic without integrating another 3rd party for markdown. We had some cases when default implementation was internal but we were asked to open it (e.g. `VideoLoading`). 

But I'm not pushing hard on it 🙂 Let's also ask what other guys think @GetStream/ios-developers

Reply:
@evsaev @tbarbugli Although having the class open is more easily customizable, this is not required to reuse our logic and still provide minor customizations. You can do this with composition:

```swift
class CustomMarkdownFormatter: MarkdownFormatter {
    let streamFormatter = DefaultMarkdownFormatter()
    
    public func containsMarkdown(text: String) -> Bool {
        return true
    }
    
    public func format(from string: String) -> NSAttributedString {
        return streamFormatter.format(string)
    }
}
```

Reply:
> Got it 👍 . Will open it then. Also will get rid of the protocol since it won’t be needed for the open class approach.

@hugobernalstream don't get rid of the `protocol` itself. That was not implied here. 
We should always prefer `composition` over `inheritance` where possible. 
Based on the comments, we're trying to go with a mixed approach, where:
1. We keep the protocol in place, so that integrator can provide his own `MarkDownFormatter` - this is great 👍
2. We turn on the inheritance on the default implementation

FYI, we took similar approach with [ChannelNameFormatter](https://github.com/GetStream/stream-chat-swift/blob/61b87c7c91fe5377c80e1a851242f1d767ea4523/Sources/StreamChatUI/Appearance%2BFormatters/ChannelNameFormatter.swift).

Reply:
This also adds a new question, is this considered a formatter? If yes, we should add this to the `appearance.formatters`. Otherwise, it doesn't make sense to name MarkdownFormatter, and having it in a different place then the formatters. Since indeed this is data formatting,  this should go to `appearance.formatters`

Reply:
Kept both approaches and moved the formatter to Appearance+Formatters as suggested.